core/silkworm/chain/config.hpp

```diff
@@ -185,29 +186,11 @@ inline constexpr ChainConfig kGoerliConfig{
     },
 };
 
-inline const ChainConfig* lookup_chain_config(uint64_t chain_id) noexcept {
-    switch (chain_id) {
-        case kMainnetConfig.chain_id:
-            return &kMainnetConfig;
-        case kRopstenConfig.chain_id:
-            return &kRopstenConfig;
-        case kRinkebyConfig.chain_id:
-            return &kRinkebyConfig;
-        case kGoerliConfig.chain_id:
-            return &kGoerliConfig;
-        default:
-            return nullptr;
-    }
-}
-
-inline uint64_t lookup_chain_id_by_name(const std::string& name) noexcept {
-    uint64_t ret{0};
-    if (iequals(name, "mainnet")) ret = kMainnetConfig.chain_id;
-    if (iequals(name, "ropsten")) ret = kRopstenConfig.chain_id;
-    if (iequals(name, "rinkeby")) ret = kRinkebyConfig.chain_id;
-    if (iequals(name, "goerli")) ret = kGoerliConfig.chain_id;
-    return ret;
-}
+//! \brief Looksup a chain config provided its network id or its common name
+const ChainConfig* lookup_chain_config(std::variant<uint64_t, std::string> identifier) noexcept;
```

Comment:
IMHO it's simpler to have two overloaded functions `lookup_chain_config(uint64_t)` and `lookup_chain_config(string_view)` instead of using  `std::variant`.

Reply:
Two overloads means two lambda functions on search. With this we can rationalize code.